Add info about conda forge manual trigger to release checklist (#8328)
Napari/napari: .github/ISSUE_TEMPLATE/release.md

- [ ] Ensure `napari/docs` successfully deploys.
- [ ] Checkout `napari/napari:main` and tag with release notes file. [See docs for details.](https://napari.org/dev/developers/coredev/release.html#tagging-the-new-release-candidate)
- [ ] Push created tag to `napari/napari`. This triggers the deployment actions.
- [ ] Check that docs deployment is successful on `napari/docs` and `napari/napari.github.io`

> - [ ] ensure [PyPI](https://pypi.org/project/napari/#history) release is out

Reviewer: ```suggestion
- [ ] Ensure [PyPI](https://pypi.org/project/napari/#history) release is out
```

> - [ ] Once PyPI is out, manually trigger the update of our Conda-forge [`napari-feedstock`](https://github.com/conda-forge/napari-feedstock) by creating an issue with title [@conda-forge-admin, please update version](https://github.com/conda-forge/napari-feedstock/issues/new?template=2-bot-commands.yml). (Alternatively, conda-forge bots will automatically make a PR, usually within 12 hours).

Reviewer: I would prefer this link to the information in the docs. My personal desire is for this to not be an "instruction manual" for release managers, but a _reminder_ for release managers **and** a nice clean checklist for community members to easily check the status. The more specificity we have here, the more we dilute the simplicity of knowledge.
```suggestion
- [ ] Conda-forge [`napari-feedstock`](https://github.com/conda-forge/napari-feedstock). **[Bot is automatic, but may be manually triggered](https://napari.org/dev/developers/coredev/release.html#new-releases)**
```

> - [ ] ensure [conda-forge release](https://anaconda.org/conda-forge/napari) is out (should be minutes after merging the feedstock PR created above).

Reviewer: ```suggestion
- [ ] Ensure [conda-forge release](https://anaconda.org/conda-forge/napari) is out (should be minutes after merging the feedstock PR created above).
```

> - [ ] Update symlink in `napari/napari.github.io` by manually triggering the [action](https://github.com/napari/napari.github.io/actions/workflows/symlink-stable.yml) (entering the new version in the field, without `v`).

Reviewer: ```suggestion
- [ ] Update symlink in `napari/napari.github.io` by manually triggering the [action](https://github.com/napari/napari.github.io/actions/workflows/symlink-stable.yml) with format "`X.Y.Z`" (no `v` prefix)
```

> - [ ] Update the `version_switcher.json` in `napari/docs`. (e.g. [this PR](https://github.com/napari/docs/pull/826))

Author: I'm a bit confused about why we need these two steps to be separated. Can't this be done in a single workflow? It's a bit clunky at the moment.

Reviewer 2: @melissawm ?

Reviewer: how would you propose doing it in a single workflow? like merging the docs PR will automatically trigger the version switcher in napari/napari.github.io? 
Yes, I do agree its clunky though

Author: Not sure; either the workflow also modifies the code/opens a PR to `napari/docs`, or viceversa triggers a symlink creation. Not a big deal, it just felt like I was doing the same task twice xD

Reviewer 3: I think the main reason is that these are tasks in two separate repos. If you prefer triggering the symlink creation automatically upon detecting a tag in napari/napari or something like that, it can be done I think. No strong feelings, it just seemed pretty low-effort at the time.

Author: Let's leave that for another time, definitely not high priority!
